modules/openapi-generator/src/main/resources/typescript-angular/api.service.mustache

```diff
@@ -130,6 +130,53 @@ export class {{classname}} {
 {{/operation}}
 {{/useHttpClient}}
 
+    private addToHttpParams(
+        httpParams: {{#useHttpClient}}HttpParams{{/useHttpClient}}{{^useHttpClient}}URLSearchParams{{/useHttpClient}},
+        value: any, key?: string): {{#useHttpClient}}HttpParams{{/useHttpClient}}{{^useHttpClient}}URLSearchParams{{/useHttpClient}}
+    {
+        if (typeof value === "object") {
+            {{#useHttpClient}}httpParams = {{/useHttpClient}}this.addToHttpParams1(httpParams, value);
+        } else {
+            {{#useHttpClient}}httpParams = {{/useHttpClient}}this.addToHttpParams1(httpParams, value, key);
+        }
+        return httpParams;
+    }
+
+    private addToHttpParams1(
+        httpParams: {{#useHttpClient}}HttpParams{{/useHttpClient}}{{^useHttpClient}}URLSearchParams{{/useHttpClient}},
+        value: any, key?: string): {{#useHttpClient}}HttpParams{{/useHttpClient}}{{^useHttpClient}}URLSearchParams{{/useHttpClient}}
+    {
+        if (typeof value === "object") {
+            if (this.isArrayLike(value)) {
+                (value as []).forEach( elem => httpParams = this.addToHttpParams1(httpParams, elem, key));
+            } else if (value instanceof Date) {
+                if (key != null) {
+                    {{#useHttpClient}}httpParams = {{/useHttpClient}}httpParams.append(key,
+                        (value as Date).toISOString().substr(0, 10));
+                } else {
+                   throw Error("key may not be null if value is Date");
+                }
+            } else {
+                Object.keys(value).forEach( k => {{#useHttpClient}}httpParams = {{/useHttpClient}}this.addToHttpParams1(
+                    httpParams, value[k], key != null ? `${key}.${k}` : k));
+            }
+        } else if (key != null) {
+            {{#useHttpClient}}httpParams = {{/useHttpClient}}httpParams.append(key, value);
+        } else {
+            throw Error("key may not be null if value is not object or array");
+        }
+        return httpParams;
+    }
+
+    private isArrayLike(obj: any): boolean {
```

Comment:
Type should be object

Reply:
No, the method should work for anything given to it (as it does now).